stats/stats-server/src/server.rs

```diff
@@ -134,3 +244,172 @@ pub async fn stats(settings: Settings) -> Result<(), anyhow::Error> {
 
     launcher::launch(&launch_settings, http_router, grpc_router).await
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
let's move tests and `async fn wait_for_blockscout_indexing(` and all related functions to sep file please, this is too much for `server.rs`

Reply:
let's name it `blockscout_waiter.rs`  🤪
![image](https://github.com/user-attachments/assets/fe3d13dc-aaf9-4015-8ec2-e703fb512856)